### PR TITLE
Dev refactor

### PR DESCRIPTION
**Pull request by EmiliaCiastek**

### CODE REVIEW

File: client/src/main/java/gui/building/FleetDropController.java [hunk 1]
```diff
@@ -150,12 +165,16 @@ private void setupShipBoardUpdater() {
...
-        fleetSender.sendFleetToServer();
-        nextButton.fireEvent(new ConnectEvent());
+        if (!fleet.isFleetEmpty()) {
+          setupClient();
+          Player player = new Player(fleet, userName.getText());
+          FleetSender fleetSender = new FleetSender(getClient(), player);
+          fleetSender.sendFleetToServer();
+          nextButton.fireEvent(new ConnectEvent());
+        } else {
+          info.setText("You can't play with empty fleet");
```
Comment (smolamarcin) [inline client/src/main/java/gui/building/FleetDropController.java]: Hardcoded value. That will be a problem in language choice system.

File: client/src/main/java/gui/playing/PlayBoardController.java [hunk 2]
```diff
@@ -94,6 +107,8 @@ private void makeMessageListenerThread() {
   }
 
   private void populateSeaWithSeaFields() {
+    sea = new Sea();
+
     for (int i = 0; i < 10; i++) {
```
Comment (smolamarcin) [inline client/src/main/java/gui/playing/PlayBoardController.java]: magic number (10)

File: client/src/main/java/gui/playing/PlayBoardController.java [hunk 3]
```diff
@@ -167,6 +184,17 @@ private void putFieldOnOpponentBoard(Field field) {
   private final EventHandler<YouHitEvent> youHit =
       event -> lastField.hit();
 
+  private final EventHandler<SunkShipEvent> shipSunk =
+      event -> {
+        String[] positionsAsString = event.getMessage().split(",");
```
Comment (smolamarcin) [inline client/src/main/java/gui/playing/PlayBoardController.java]: Maybe you should extract "," as a constant?

File: server/src/main/java/engine/ActiveGame.java [hunk 4]
```diff
@@ -40,7 +41,14 @@ public GameRunnerState run() throws IOException {
       ShotResult result = round.fireShot(fleetUnderFire, shot);
       logShotInfo(shot, result);
 
-      if (result == ShotResult.HIT) {
+      if (result == ShotResult.SUNK) {
```
Comment (smolamarcin) [inline server/src/main/java/engine/ActiveGame.java]: It does not look good.

File: server/src/main/java/engine/HitChecker.java [hunk 5]
```diff
@@ -22,6 +22,11 @@ public ShotResult checkShot(final int position) {
 
     if (!isHitAgain && isShipPosition) {
       fleet.hit(position);
+      boolean isShipSunk = fleet.getHitFields().containsAll(fleet.getShipByPosition(position).getFields());
```
Comment (smolamarcin) [inline server/src/main/java/engine/HitChecker.java]: train... (get get)

File: server/src/main/java/fleet/CustomFleet.java [hunk 6]
```diff
@@ -35,4 +38,13 @@ public void hit(final int position) {
   public List<Integer> getHitFields() {
     return hitFields;
   }
+
+  public ShipModel getShipByPosition(Integer positionToSearch) {
+    Optional<ShipModel> foundShip = ships.stream()
```
Comment (smolamarcin) [inline server/src/main/java/fleet/CustomFleet.java]: 👍

File: utils/src/main/java/model/ShipModel.java [hunk 7]
```diff
@@ -31,4 +31,13 @@ public int getNumberOfMasts() {
   public List<Integer> getFields() {
     return fields;
   }
+
+  @Override
+  public String toString() {
+    StringBuilder builder = new StringBuilder();
+
```
Comment (smolamarcin) [inline utils/src/main/java/model/ShipModel.java]: Empty lines

File: client/src/main/java/gui/building/FleetDropController.java [hunk 8]
```diff
@@ -150,12 +165,16 @@ private void setupShipBoardUpdater() {
 
   private final EventHandler<MouseEvent> connectWhenClicked =
       event -> {
-        connectButton.setVisible(false);
-        setupClient();
-        Player player = new Player(fleet , userName.getText());
-        FleetSender fleetSender = new FleetSender(getClient(), player);
-        fleetSender.sendFleetToServer();
-        nextButton.fireEvent(new ConnectEvent());
+        if (!fleet.isFleetEmpty()) {
```
Comment (smolamarcin) [inline client/src/main/java/gui/building/FleetDropController.java]: Only to improve readability: extract !fleet.isFleetEmpty() to method.

File: client/src/main/java/gui/playing/PlayBoardController.java [hunk 9]
```diff
@@ -102,6 +117,8 @@ private void populateSeaWithSeaFields() {
           field.marked();
           lastField = field;
         });
+
```
Comment (smolamarcin) [inline client/src/main/java/gui/playing/PlayBoardController.java]: empty line

File: client/src/main/java/model/MastBoundariesPositions.java [hunk 10]
```diff
@@ -28,8 +26,8 @@
 
     for (int neighbourColumnValue = -1; neighbourColumnValue <= 1; neighbourColumnValue++) {
```
Comment (smolamarcin) [inline client/src/main/java/model/MastBoundariesPositions.java]: Magic numbers

File: server/src/main/java/engine/ActiveGame.java (hunk 4 above)
Comment (EmiliaCiastek) [reply server/src/main/java/engine/ActiveGame.java]: We still don't know how to refactor and test this class

File: client/src/main/java/model/MastBoundariesPositions.java (hunk 10 above)
Comment (EmiliaCiastek) [reply client/src/main/java/model/MastBoundariesPositions.java]: step variable added

File: server/src/main/java/engine/HitChecker.java (hunk 5 above)
Comment (EmiliaCiastek) [reply server/src/main/java/engine/HitChecker.java]: fixed

File: utils/src/main/java/model/ShipModel.java (hunk 7 above)
Comment (EmiliaCiastek) [reply utils/src/main/java/model/ShipModel.java]: fixed

File: client/src/main/java/gui/playing/PlayBoardController.java (hunk 3 above)
Comment (EmiliaCiastek) [reply client/src/main/java/gui/playing/PlayBoardController.java]: done

File: client/src/main/java/gui/playing/PlayBoardController.java (hunk 9 above)
Comment (EmiliaCiastek) [reply client/src/main/java/gui/playing/PlayBoardController.java]: I think it's more readable with empty line

File: client/src/main/java/gui/playing/PlayBoardController.java (hunk 2 above)
Comment (EmiliaCiastek) [reply client/src/main/java/gui/playing/PlayBoardController.java]: done

File: client/src/main/java/gui/building/FleetDropController.java (hunk 1 above)
Comment (EmiliaCiastek) [reply client/src/main/java/gui/building/FleetDropController.java]: done

File: client/src/main/java/gui/playing/PlayBoardController.java (hunk 9 above)
Comment (smolamarcin) [reply client/src/main/java/gui/playing/PlayBoardController.java]: 👍

**Pull request merged**